### PR TITLE
Add GlowErrs to runtime

### PR DESCRIPTION
**Pull request by jackm321**
*Description*:
Return `GlowErr`s from methods in DeviceManager, Provisioner, HostManager, Executor, etc and handle them in onnxifi.

*Testing*:
`Ninja check`

*Documentation*:
Comments

### CODE REVIEW

**Comment by jackm321** [comment]
This only 1% done but there will be quite a few places to change here so I'm putting this up now to get opinions on the general direction.

**Comment by jackm321** [comment]
Actually it's about 50% done now, it's not too much churn

**Comment by jackm321** [comment]
Ok that's all the changes I want to do right now. Changing `ResultCBTy` involves a huge ton of changes to executor that I'm going to hold off on.

File: tests/unittests/DeviceManagerTest.cpp [hunk 1]
```diff
@@ -76,29 +76,33 @@ std::pair<std::promise<ResultType>, std::future<ResultType>> getFutureHelper() {
...
 TEST_P(DeviceManagerTest, Basic) {
   auto module = makeBasicModule();
   std::vector<std::unique_ptr<CompiledFunction>> backing;
   FunctionMapTy functions =
       compileFunctions(backendKind, module.get(), backing);
 
   auto *device = DeviceManager::createDeviceManager(backendKind);
-  ResultCode initResult = device->init();
-  EXPECT_EQ(initResult, ResultCode::Executed);
+  EXIT_ON_ERR(device->init());
```
Comment (gcatron) [inline tests/unittests/DeviceManagerTest.cpp]: Does exit_on_err end the entire bank of unit tests?

File: lib/ExecutionEngine/ExecutionEngine.cpp [hunk 2]
```diff
@@ -168,15 +166,16 @@ void ExecutionEngine::insertCompiledFunction(
   functionMap[name] = func.get();
   compiledFunctions_[name] = std::move(func);
 
-  std::promise<runtime::ResultCode> addPromise;
+  // TODO: make this a std::promise<llvm::Error>
+  std::promise<bool> addPromise;
```
Comment (rdzhabarov) [inline lib/ExecutionEngine/ExecutionEngine.cpp]: qq: why not make it `std::promise<llvm::Error>` now?

Comment (jackm321) [reply lib/ExecutionEngine/ExecutionEngine.cpp]: Oops good catch thanks! That's what the todo was for. I'll do it now

**Pull request merged**